backend/dataall/modules/dataset_sharing/services/share_managers/lf_share_manager.py

```diff
@@ -480,17 +480,21 @@ def handle_share_failure(
         -------
         True if alarm published successfully
         """
-        logging.error(
-            f'Failed to share table {table.GlueTableName} '
-            f'from source account {self.source_environment.AwsAccountId}//{self.source_environment.region} '
-            f'with target account {self.target_environment.AwsAccountId}/{self.target_environment.region}'
-            f'due to: {error}'
-        )
+        try:
+            logging.error(
+                f'Failed to share table {table.GlueTableName} '
+                f'from source account {self.source_environment.AwsAccountId}//{self.source_environment.region} '
+                f'with target account {self.target_environment.AwsAccountId}/{self.target_environment.region}'
+                f'due to: {error}'
+            )
 
-        DatasetAlarmService().trigger_table_sharing_failure_alarm(
-            table, self.share, self.target_environment
-        )
-        return True
+            DatasetAlarmService().trigger_table_sharing_failure_alarm(
+                table, self.share, self.target_environment
+            )
+            return True
+        except Exception as e:
+            logger.error(f"Exception during handle_share_failure: {e}")
```

Comment:
This could be more meaningful: "Could not process dataset alarms" because otherwise this exception message is useless because we will see where it happened in the exception stack trace.. Also there's a nicer way to do it like this:

logger.error("Could not process dataset alarms: " , exc_info=True)

Reply:
Alright!